File: 20-valid-parentheses/valid-parentheses.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isValid = function(s) {
    const stk=[]

    for (let i=0;i<s.length;i++)
    {
        if (s[i]==='('||s[i]==='{'||s[i]==='[')
        {
            stk.push(s[i]);
        }

        else {
            let last=stk.pop();
            
            if (
             (s[i]===')'&&last!=='(') ||
             (s[i]==='}'&&last!=='{') || 
             (s[i]===']'&&last!=='[') )
            {
            return false; 
            }
        }

    }
        return stk.length===0;

};

File: 20-valid-parentheses/valid-parentheses.java
class Solution {
    public boolean isValid(String s) {

        Stack <Character> st=new Stack<>();
        
        for (int i=0;i<s.length();i++)
        {
            char a=s.charAt(i);
            if(a=='('|| a=='[' || a=='{')
            {
                st.push(a);
            }
            else {
                if(!st.isEmpty())
                {
                    char x=st.pop();
                     if(x=='(' && a!=')' ||
                        x=='[' && a!=']' ||
                        x=='{' && a!='}' )
                        {
                            return false;
                        }

                }
               else return false;
            }
        }
        return st.isEmpty();
    }
}